src/Adapter/MSTest.TestAdapter/MSTestDiscoverer.cs

```diff
@@ -51,6 +52,14 @@ public class MSTestDiscoverer : ITestDiscoverer
         try
         {
             MSTestSettings.PopulateSettings(discoveryContext);
+
+#if !NETFRAMEWORK
+            MSTestAdapterSettings settings = MSTestSettingsProvider.Settings;
+            if (settings.SearchDirectories.Count > 1)
+            {
+                logger.SendMessage(TestMessageLevel.Warning, Resource.AssemblyResolutionIsOnlyWorkingWithNetFramework);
+            }
+#endif
```

Comment:
I think this needs to be done also in `MSTestExecutor`. I would probably also hide this details under the `MSTestSettings` class by adding a `ValidateSettings` method so that it's easier to reuse and it's grouping settings validation under the same umbrella (in case we need to validate more entries in the future).

Reply:
A `bool MSTestAdapterSettings.Validate(Action<string> validationMessage)` maybe could help